query/src/main/scala/filodb/query/exec/MultiSchemaPartitionsExec.scala

```diff
@@ -39,11 +41,45 @@ final case class MultiSchemaPartitionsExec(queryContext: QueryContext,
   @transient // dont serialize the SelectRawPartitionsExec plan created for plan execution
   var finalPlan: SelectRawPartitionsExec = _
 
+  // Remove _columnName from metricColumn and generate PartLookupResult
+  private def removeColumn(filters: Seq[ColumnFilter], metricName: String,
+                           columnName: String, source: ChunkSource,
+                           querySession: QuerySession) = {
+    // Assume metric name has only equal filter
+    val filterWithoutColumn = filters.filterNot(_.column == metricColumn) :+
+      ColumnFilter(metricColumn, Equals(metricName.replace(s"_$columnName", "")))
```

Comment:
`metricName.replace(s"_$columnName", "")` should not be used since it can also replace middle portions. Use `stripSuffix` method.